zerver/views/user_groups.py

```diff
@@ -141,22 +147,40 @@ def edit_user_group(
         if setting_group_id_name not in request_settings_dict:  # nocoverage
             continue
 
-        if request_settings_dict[setting_group_id_name] is not None and request_settings_dict[
-            setting_group_id_name
-        ] != getattr(user_group, setting_group_id_name):
-            setting_value_group_id = request_settings_dict[setting_group_id_name]
-            setting_value_group = access_user_group_for_setting(
-                setting_value_group_id,
-                user_profile,
-                setting_name=setting_name,
-                require_system_group=permission_config.require_system_group,
-                allow_internet_group=permission_config.allow_internet_group,
-                allow_owners_group=permission_config.allow_owners_group,
-                allow_nobody_group=permission_config.allow_nobody_group,
-            )
-            do_change_user_group_permission_setting(
-                user_group, setting_name, setting_value_group, acting_user=user_profile
-            )
+        if request_settings_dict[setting_group_id_name] is not None:
+            value_changed = False
+            current_value = getattr(user_group, setting_name)
+            if (
+                isinstance(request_settings_dict[setting_group_id_name], str)
+                and request_settings_dict[setting_group_id_name] != current_value.name
+            ):
+                value_changed = True
+            if (
+                isinstance(request_settings_dict[setting_group_id_name], int)
+                and request_settings_dict[setting_group_id_name] != current_value.id
+            ):
+                value_changed = True
```

Comment:
So, is it final that we want to keep this changes? I hope you have looked at second point of my comment [here](https://github.com/zulip/zulip/pull/26355#issuecomment-1680992834). Currently, there is no scenario in which we are sending `string` to server from frontend for changing a group level setting. So, this change is not used  presently. I did because earlier I misunderstood the design.

Reply:
I am not sure whether we would have such case as I am not sure how would be the UI in future. Will we directly provide single users in the options, or we would provide a UI for creating a new group, where ultimately a single user group will be created and its ID will be sent to the server.

We can decide these once we have feedback from Tim on other comments.